### DISCUSSION TITLE
Avro schemas with references

### DISCUSSION THREAD

**Discussion by CROprogrammer** [sngular/kloadgen#396] [Q&A]
Hi,

I was trying to test if Avro schema with a reference to other schema can be loaded to KLoadGen from schema registry, but unfortunately it cannot.

I was using schema Student:
```
{
  "type": "record",
  "namespace": "com.example",
  "name": "Student",
  "fields": [
    {
      "name": "name",
      "type": "string"
    },
    {
      "name": "age",
      "type": "int",
      "default": -1
    }
  ]
}
```
and schema Address that has reference to schema Student:
```
{
  "type": "record",
  "namespace": "com.example",
  "name": "Address",
  "fields": [
    {
      "name": "name",
      "type": "string"
    },
    {
      "name": "student",
      "type": "com.example.Student"
    }
  ]
}
```

I kept getting error message:
`org.apache.avro.SchemaParseException: "com.example.Student" is not a defined name. The type of the "student" field must be a defined name or a {"type": ...} expression.
`

Are you planning to add support for that by any chance?

**Comment by jemacineiras** [comment]
Hi @CROprogrammer,

for now that functionality is only supported for ProtoBuff schemas as far as I know. If you please can open a Issue related to that feature we will start working asap on it.

Please include all information about your setup and requirements on that.

Cheers

**Reply by CROprogrammer** [reply]
Will do!

Confluent/Apicurio SerDes have support for schema with references, so it would be handy feature.

**Reply by jemacineiras** [reply]
Cool, problem here is not the ser/des part. Is analyze the schemas and extract the information to offer you how you want to configure the Test Plan. We are doing that manually and processing all the information. But we can do it.

**Comment by CROprogrammer** [comment]
https://github.com/sngular/kloadgen/issues/397
